RollupDocumentationIT.testStartRollupJob fails

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=debian/42/console

```
12:57:00 ERROR   0.36s | RollupDocumentationIT.testStartRollupJob <<< FAILURES!
12:57:00    > Throwable #1: org.elasticsearch.client.ResponseException: method [DELETE], host [http://[::1]:52535], URI [/_xpack/rollup/job/job_1], status line [HTTP/1.1 500 Internal Server Error]
12:57:00    > {"task_failures":[{"task_id":8905,"node_id":"FhzPJ_ibQoS6vX9AbmETzA","status":"INTERNAL_SERVER_ERROR","reason":{"type":"illegal_state_exception","reason":"Could not delete job [job_1] because indexer state is [STOPPING].  Job must be [STOPPED] before deletion."}}],"acknowledged":false}
12:57:00    > 	at __randomizedtesting.SeedInfo.seed([8485FBE4F308CC81:EB88BE3030DD61E3]:0)
12:57:00    > 	at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:687)
12:57:00    > 	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:218)
12:57:00    > 	at org.elasticsearch.client.documentation.RollupDocumentationIT.deleteRollupJobs(RollupDocumentationIT.java:437)
12:57:00    > 	at org.elasticsearch.client.documentation.RollupDocumentationIT.wipeRollup(RollupDocumentationIT.java:417)
12:57:00    > 	at java.lang.Thread.run(Thread.java:748)
```

Reproduce line (though does not repro locally):

```
REPRODUCE WITH: ./gradlew :client:rest-high-level:integTestRunner -Dtests.seed=8485FBE4F308CC81 -Dtests.class=org.elasticsearch.client.documentation.RollupDocumentationIT -Dtests.method="testStartRollupJob" -Dtests.security.manager=true -Dtests.locale=ru-RU -Dtests.timezone=Africa/Malabo -Dcompiler.java=11 -Druntime.java=8
```

This resurfaced again on 6.x. Seems to be a timeout issue.
 Does NOT reproduce locally on 6.x for me.

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+intake/762/console

```
REPRODUCE WITH: ./gradlew :client:rest-high-level:integTestRunner \
  -Dtests.seed=F86D6F8A86509455 \
  -Dtests.class=org.elasticsearch.client.documentation.RollupDocumentationIT \
  -Dtests.method="testStartRollupJob" \
  -Dtests.security.manager=true \
  -Dtests.locale=tr \
  -Dtests.timezone=Antarctica/South_Pole \
  -Dcompiler.java=11 \
  -Druntime.java=8

REPRODUCE WITH: ./gradlew :client:rest-high-level:integTestRunner \
  -Dtests.seed=F86D6F8A86509455 \
  -Dtests.class=org.elasticsearch.client.documentation.RollupDocumentationIT \
  -Dtests.method="testDeleteRollupJob" \
  -Dtests.security.manager=true \
  -Dtests.locale=tr \
  -Dtests.timezone=Antarctica/South_Pole \
  -Dcompiler.java=11 \
  -Druntime.java=8

REPRODUCE WITH: ./gradlew :client:rest-high-level:integTestRunner \
  -Dtests.seed=F86D6F8A86509455 \
  -Dtests.class=org.elasticsearch.client.documentation.RollupDocumentationIT \
  -Dtests.method="testDeleteRollupJob" \
  -Dtests.security.manager=true \
  -Dtests.locale=tr \
  -Dtests.timezone=Antarctica/South_Pole \
  -Dcompiler.java=11 \
  -Druntime.java=8

REPRODUCE WITH: ./gradlew :client:rest-high-level:integTestRunner \
  -Dtests.seed=F86D6F8A86509455 \
  -Dtests.class=org.elasticsearch.client.documentation.RollupDocumentationIT \
  -Dtests.method="testGetRollupCaps" \
  -Dtests.security.manager=true \
  -Dtests.locale=tr \
  -Dtests.timezone=Antarctica/South_Pole \
  -Dcompiler.java=11 \
  -Druntime.java=8
```
Relevant stacktrace:
```
17:46:20     > {"error":{"root_cause":[{"type":"timeout_exception","reason":"Timed out after [10s] while waiting for rollup job [job_1] to stop"}],"type":"timeout_exception","reason":"Timed out after [10s] while waiting for rollup job [job_1] to stop"},"status":500}
17:46:20     >  at __randomizedtesting.SeedInfo.seed([F86D6F8A86509455:97602A5E45853937]:0)
17:46:20     >  at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:929)
17:46:20     >  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:229)
17:46:20     >  at org.elasticsearch.test.rest.ESRestTestCase.wipeRollupJobs(ESRestTestCase.java:570)
17:46:20     >  at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:482)
17:46:20     >  at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:263)
17:46:20     >  at java.lang.Thread.run(Thread.java:748)
17:46:20     > Caused by: org.elasticsearch.client.ResponseException: method [POST], host [http://[::1]:38195], URI [/_xpack/rollup/job/job_1/_stop?wait_for_completion=true&timeout=10s], status line [HTTP/1.1 500 Internal Server Error]
17:46:20     > {"error":{"root_cause":[{"type":"timeout_exception","reason":"Timed out after [10s] while waiting for rollup job [job_1] to stop"}],"type":"timeout_exception","reason":"Timed out after [10s] while waiting for rollup job [job_1] to stop"},"status":500}
17:46:20     >  at org.elasticsearch.client.RestClient$1.completed(RestClient.java:548)
17:46:20     >  at org.elasticsearch.client.RestClient$1.completed(RestClient.java:533)
17:46:20     >  at org.apache.http.concurrent.BasicFuture.completed(BasicFuture.java:119)
17:46:20     >  at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseCompleted(DefaultClientExchangeHandlerImpl.java:177)
17:46:20     >  at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.processResponse(HttpAsyncRequestExecutor.java:436)
17:46:20     >  at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.inputReady(HttpAsyncRequestExecutor.java:326)
17:46:20     >  at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:265)
17:46:20     >  at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:81)
17:46:20     >  at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:39)
17:46:20     >  at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114)
17:46:20     >  at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162)
17:46:20     >  at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337)
17:46:20     >  at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
17:46:20     >  at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276)
17:46:20     >  at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
17:46:20     >  at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:588)
```

--- a/docs/reference/rollup/apis/stop-job.asciidoc
+++ b/docs/reference/rollup/apis/stop-job.asciidoc
@@ -56,6 +56,7 @@ POST _rollup/job/sensor/_stop
 --------------------------------------------------
 // CONSOLE
 // TEST[setup:sensor_started_rollup_job]
+// TEST[s/_stop/_stop?wait_for_completion=true&timeout=10s/]
 
 Which will return the response:
 

--- a/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java
@@ -460,7 +460,7 @@ public abstract class ESRestTestCase extends ESTestCase {
     private void wipeCluster() throws Exception {
 
         // Cleanup rollup before deleting indices.  A rollup job might have bulks in-flight,
-        // so if we need to fully shut them down first otherwise a job might stall waiting
+        // so we need to fully shut them down first otherwise a job might stall waiting
         // for a bulk to finish against a non-existing index (and then fail tests)
         if (hasXPack && false == preserveRollupJobsUponCompletion()) {
             wipeRollupJobs();

--- a/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java
@@ -458,6 +458,15 @@ public abstract class ESRestTestCase extends ESTestCase {
     }
 
     private void wipeCluster() throws Exception {
+
+        // Cleanup rollup before deleting indices.  A rollup job might have bulks in-flight,
+        // so if we need to fully shut them down first otherwise a job might stall waiting
+        // for a bulk to finish against a non-existing index (and then fail tests)
+        if (hasXPack && false == preserveRollupJobsUponCompletion()) {
+            wipeRollupJobs();
+            waitForPendingRollupTasks();
+        }
+
         if (preserveIndicesUponCompletion() == false) {
             // wipe indices
             try {
@@ -503,11 +512,6 @@ public abstract class ESRestTestCase extends ESTestCase {
         // wipe cluster settings
         if (preserveClusterSettings() == false) {
             wipeClusterSettings();
-        }
-
-        if (hasXPack && false == preserveRollupJobsUponCompletion()) {
-            wipeRollupJobs();
-            waitForPendingRollupTasks();
         }
 
         if (hasXPack && false == preserveILMPoliciesUponCompletion()) {

